crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -121,35 +121,37 @@ use std::{
 ///
 /// [`SyncCell`]: bevy_utils::synccell::SyncCell
 /// [`Exclusive`]: https://doc.rust-lang.org/nightly/std/sync/struct.Exclusive.html
-pub trait SystemParam: Sized {
-    type State: SystemParamState;
-}
-
-pub type SystemParamItem<'w, 's, P> = <<P as SystemParam>::State as SystemParamState>::Item<'w, 's>;
-
-/// The state of a [`SystemParam`].
 ///
 /// # Safety
 ///
 /// It is the implementor's responsibility to ensure `system_meta` is populated with the _exact_
```

Comment:
This could use more pointers to where `system_meta` is stored (ideally a link, so we're notified if it gets moved).